src/main/java/com/bugsnag/HandledState.java

```diff
@@ -21,31 +25,53 @@ public String toString() {
     }
 
     private final SeverityReasonType severityReasonType;
+    private final Map<String, String> severityReasonAttributes;
     private final Severity originalSeverity;
     private final boolean unhandled;
 
     private Severity currentSeverity;
 
-    static HandledState newInstance(SeverityReasonType severityReasonType) {
-        return newInstance(severityReasonType, null);
+    static HandledState newInstance(SeverityReasonType severityReason) {
+        return newInstance(severityReason, null, null, false);
+    }
+
+    static HandledState newInstance(SeverityReasonType severityReason,
+                                    Map<String, String> severityReasonAttributes) {
+        return newInstance(severityReason, severityReasonAttributes, null, false);
     }
 
-    static HandledState newInstance(SeverityReasonType severityReasonType, Severity severity) {
+    static HandledState newInstance(SeverityReasonType severityReason, Severity severity) {
+        return newInstance(severityReason, null, severity, false);
```

Comment:
It'd be nice to consider passing in `Collections.emptyMap` so we can assume the field is always non-null. The getter could be annotated with a setter to prevent an empty object being serialised

Reply:
We're already making sure that only non-empty fields are serialized:
https://github.com/bugsnag/bugsnag-java/blob/b17d7f3c7fee3e12a913f90bf86b7a947ea85ee4/src/main/java/com/bugsnag/serialization/Serializer.java#L20